unit-tests/src/test/java/org/eclipse/collections/impl/collector/Collectors2Test.java

```diff
@@ -1111,6 +1112,34 @@ public void countByParallel()
         Assert.assertEquals(50000, counts.occurrencesOf(1));
     }
 
+    @Test
+    public void countByEach()
+    {
+        List<Interval> intervals = FastList.newListWith(
+                Interval.evensFromTo(1, 100),
+                Interval.oddsFromTo(1, 100));
+
+        MutableBag<Integer> counts = intervals.stream().collect(Collectors2.countByEach(iv -> iv.collect(i -> i % 2).toList()));
```

Comment:
I think the call to toList() should be unnecessary but otherwise this looks good.